cats_frontend/src/app/modules/cats/pages/cats-list/cats-list.component.ts

```diff
@@ -59,144 +61,117 @@ export class CatsListComponent implements OnInit {
-            cat.comments = comments;
-        });
+    // called when user is expanding a cat
+    expand(cat: ICat): void {
+        // dispatch the load action
+        this.store.dispatch(CommentsActions.load({ id: cat.id! }));
     }
 
-    // add or remove query param of the page
-    private updateQueryParam() {
-        if(this.first === 0) {
-          this.router.navigate([], {
-            relativeTo: this.route,
-            queryParams: { page: null},
-            queryParamsHandling: 'merge',
-          });
-        } else {
-          this.router.navigate([], {
-            relativeTo: this.route,
-            queryParams: { page: this.first / this.rows + 1 },
-            queryParamsHandling: 'merge',
-          });
-        }
-    }
 
-    goToCreate(): void {
-        this.router.navigate(['cats/create']);
-    }
+    // ** Cats Methods **
+    updateCat(): void {
+        const cat: ICat = {
+            id: this.editCatForm.get('id')?.value,
+            name: this.editCatForm.get('name')?.value,
+            breed: this.editCatForm.get('breed')?.value,
+            // return birthday as a string dd-mm-yyyy
+            birthday: this.datePipe.transform(this.editCatForm.get('birthday')?.value, 'YYYY-MM-dd')!,
+            description: this.editCatForm.get('description')?.value,
+          };
 
-    deleteCat(cat: ICat): void {
-        this.catsService.delete(cat.id!).subscribe(() => {
-            this.cats = this.cats.filter(c => c.id !== cat.id);
-            this.messageService.add({ severity: 'success', summary: 'Success', detail: 'The cat is deleted' });
-        });
-    }
+        // dispatch the update action
+        this.store.dispatch(CatsActions.update({ cat }));
 
-    startCatEdit(cat: ICat): void {
-        this.editCatForm.setValue({
-            id: cat.id,
-            name: cat.name,
-            breed: cat.breed,
-            birthday: new Date(cat.birthday),
-            description: cat.description
-        });
-      
-        this.editSidebarIsOpened = true;
+        // close the sidebar
+        this.hideOrShowSideBar();
     }
 
+    deleteCat(cat: ICat): void {
 
-    cancelCatEdit() { 
-        this.editSidebarIsOpened = false;
+        // dispatch the delete action
+        this.store.dispatch(CatsActions.delete({ id: cat.id! }));
     }
 
+    // ** Comments Methods **
+
     startCommentCreate(cat: ICat): void {
        const dialogRef = this.dialogService.open(CreateCommentComponent, {
-        data: {
-            cat: cat,
-        },
-        header: 'Create a comment',
-        width: '50%'
+            data: {
+                cat: cat,
+            },
+            header: 'Create a comment',
+            width: '50%'
        })
 
        // subscribe to the close event of the modal 
        // and update the comment returned 
         dialogRef.onClose.subscribe((comment: IComment) => {
-            for(let c of this.cats) {
```

Comment:
remove this logic and only dispatch an add comment action if the modal return an comment